src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -291,86 +425,86 @@ def find_fixed_range(testcase_id, job_type):
   if max_index is None:
-    raise errors.BuildNotFoundError(max_revision, job_type)
+    error_message = f'Build {max_revision} no longer exists.'
+    return uworker_io.UworkerOutput(
+        testcase=testcase,
+        error_message=error_message,
+        error=uworker_msg_pb2.ErrorType.PROGRESSION_BUILD_NOT_FOUND)
 
   testcase = data_handler.get_testcase_by_id(testcase_id)
-  data_handler.update_testcase_comment(testcase, data_types.TaskState.STARTED,
-                                       'r%d' % max_revision)
 
   # Check to see if this testcase is still crashing now. If it is, then just
   # bail out.
-  result = _testcase_reproduces_in_revision(
+  result, error = _testcase_reproduces_in_revision(
       testcase,
       testcase_file_path,
       job_type,
       max_revision,
       update_metadata=True)
+  if error is not None:
+    return error
+
   if result.is_crash():
-    logs.log('Found crash with same signature on latest revision r%d.' %
-             max_revision)
+    logs.log(f'Found crash with same signature on latest'
+             f' revision r{max_revision}.')
     app_path = environment.get_value('APP_PATH')
     command = testcase_manager.get_command_line_for_application(
         testcase_file_path, app_path=app_path, needs_http=testcase.http_flag)
     symbolized_crash_stacktrace = result.get_stacktrace(symbolized=True)
     unsymbolized_crash_stacktrace = result.get_stacktrace(symbolized=False)
     stacktrace = utils.get_crash_stacktrace_output(
         command, symbolized_crash_stacktrace, unsymbolized_crash_stacktrace)
-    testcase = data_handler.get_testcase_by_id(testcase_id)
-    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
-        stacktrace)
-    data_handler.update_progression_completion_metadata(
-        testcase,
-        max_revision,
-        is_crash=True,
-        message='still crashes on latest revision r%s' % max_revision)
-
-    # Since we've verified that the test case is still crashing, clear out any
-    # metadata indicating potential flake from previous runs.
-    task_creation.mark_unreproducible_if_flaky(testcase, False)
 
-    # For chromium project, save latest crash information for later upload
-    # to chromecrash/.
     state = result.get_symbolized_data()
-    crash_uploader.save_crash_info_if_needed(testcase_id, max_revision,
-                                             job_type, state.crash_type,
-                                             state.crash_address, state.frames)
-    return
+    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
+        stacktrace)
 
+    error_message = f'Still crashes on latest revision r{max_revision}.'
+    serialized_crash_stack_frames = crash_uploader.get_symbolized_stack_bytes(
+        state.crash_type, state.crash_address, state.frames)
+    progression_task_output = uworker_io.ProgressionTaskOutput(
+        crash_on_latest=True,
```

Comment:
Can you make a crash_on_latest_message field and use that instead of (mis)using error message please.